src/zope/schema/tests/test__bootstrapfields.py

```diff
@@ -752,9 +754,29 @@ def test_fromUnicode_hit(self):
         self.assertEqual(txt.fromUnicode(u'True'), True)
         self.assertEqual(txt.fromUnicode(u'true'), True)
 
+class ConformanceMixin(object):
 
-class OrderableMissingValueMixin(object):
+    def _getTargetClass(self):
+        raise NotImplementedError
+
+    def _getTargetInterface(self):
+        raise NotImplementedError
+
+    def _makeOne(self, *args, **kwargs):
+        return self._getTargetClass()(*args, **kwargs)
+
+    def test_class_conforms_to_iface(self):
+        from zope.interface.verify import verifyClass
+        verifyClass(self._getTargetInterface(), self._getTargetClass())
+        return verifyClass
+
+    def test_instance_conforms_to_iface(self):
+        from zope.interface.verify import verifyObject
+        verifyObject(self._getTargetInterface(), self._makeOne())
+        return verifyObject
 
+
+class OrderableMissingValueMixin(ConformanceMixin):
```

Comment:
But then I would need to repeat myself in this class definition (defining the `_getTargetClass` and `_makeOne` methods) and in every test case that extends `OrderableMissingValueMixin` (to also extend `ConformanceMixin`). 

I was taking the same approach with the `EqualityTestsMixin` that #51 introduced (making it extend `ConformanceMixin`) and adding it to essentially all the field test classes, so that would have to be three things to subclass.

Would you prefer a base unittest class that extends both of the mixins? (Except I think the testrunner complains if it finds a unittest class that has no tests.)

Reply:
Creating a `BasicFieldTests`, say, that is the contents of `ConformanceMixin` and `EqualityTestsMixin` and subclasses `TestCase` and has concrete implementations of `_getTargetClass() -> Field` and `_getTargetInterface() -> IField`, and then having everything subclass that could work. But if you neglect to override those two methods in a subclass, you wouldn't actually be testing what you think you're testing.

I suppose they could be written to test `type(self)` and if it's not `BasicFieldTests` they would raise, but that seems ugly compared to just leaving them abstract. (Abstract classes are detected by linters, whereas that would just be a runtime error.)

Reply:
I would prefer to mix in *both* `OrderableMissingValueMixin` and `ConformanceMixin` whereever appropriate.

Reply:
OK.